File: lib/src/main/java/org/simplepresenter/PresenterHolder.java
package org.simplepresenter;

import java.util.HashMap;
import java.util.Map;

public final class PresenterHolder {

    public static final PresenterHolder INSTANCE = new PresenterHolder();

    private PresenterHolder() {
    }

    private Map<Integer, Presenter> map = new HashMap<>();

    public void registerPresenter(Presenter presenter) {
        map.put(presenter.hashCode(), presenter);
    }


    public Presenter findPresenterById(int id) {
        return map.get(id);
    }

    public void removePresenterById(int id) {
        map.remove(id);
    }
}

File: lib/src/main/java/org/simplepresenter/PresenterHolder.kt
package org.simplepresenter

import java.util.*

internal object PresenterHolder {

    private val map = HashMap<Int, Presenter>()

    fun registerPresenter(presenter: Presenter) {
        map.put(presenter.hashCode(), presenter)
    }

    fun findPresenterById(id: Int) = map[id]

    fun removePresenterById(id: Int) = map.remove(id)
}